src/context.rs

```diff
@@ -26,12 +26,12 @@ use ffi::enums::{Status, Antialias, LineCap, LineJoin, FillRule};
 use ::patterns::{wrap_pattern, Pattern};
 use surface::Surface;
 
-pub struct RectangleVec {
+pub struct RectangleVec<'a> {
     ptr: *mut cairo_rectangle_list_t,
-    pub rectangles: CVec<Rectangle>,
+    pub rectangles: &'a[Rectangle], // Vec<Rectangle>,
 }
```

Comment:
Line `pub fn copy_clip_rectangle_list(&self) -> RectangleVec {` with your version of `RectangleVec` actually mean `pub fn copy_clip_rectangle_list<'a>(&'a self) -> RectangleVec<'a> {`
So borrow checker think that `RectangleVec` can't outlive `Context` object for that function was called.
In most cases is not problem, but if you need store this vector for long time borrow checker nags about lifetimes.
PS. No, `RectangleVec` instance can't be copied, as it don't realize `Clone, Copy` in cairo crate, and has private member, that disallow create objects of this type outside